src/libcore/slice.rs

```diff
@@ -1698,12 +1782,47 @@ impl<T: PartialOrd> PartialOrd for [T] {
         let rhs = &other[..l];
 
         for i in 0..l {
-            match lhs[i].partial_cmp(&rhs[i]) {
-                Some(Ordering::Equal) => (),
+            match lhs[i].cmp(&rhs[i]) {
+                Ordering::Equal => (),
                 non_eq => return non_eq,
             }
         }
 
-        self.len().partial_cmp(&other.len())
+        self.len().cmp(&other.len())
+    }
+}
+
+// memcmp compares a sequence of unsigned bytes lexicographically.
+// this matches the order we want for [u8], but no others (not even [i8]).
```

Comment:
Couldn't we implement this for `[i8]`? We may not be able to implement it for larger types due to endianness (unless you're on a big-endian platform), but I'm drawing a blank on why to *not* implement it for `i8`. The semantics for comparing `[u8]` are the same with this and the above impl, right?